packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.spec.tsx

```diff
@@ -564,9 +564,19 @@ describe('HttpOperation', () => {
 });
 
 function getDeprecatedBadge() {
-  return screen.queryByRole('badge', { name: /Deprecated/i });
+  const badges = screen.getAllByRole('badge');
+  return screen.queryByText(
+    (_, element) => element !== null && badges.includes(element as HTMLElement) && element.textContent === 'Deprecated',
+  );
 }
 
 function getSecurityBadge(re: RegExp) {
-  return screen.queryByRole('badge', { name: re });
+  const badges = screen.getAllByRole('badge');
+  return screen.queryByText(
+    (_, element) =>
+      element !== null &&
+      element.textContent !== null &&
+      badges.includes(element as HTMLElement) &&
+      re.test(element.textContent),
+  );
```

Comment:
Why does this need to be so complicated? Won't the original do? If it's a mosaic accessibility problem, can we fix it?

Reply:
Because the `badge` role is on the parent div and it `findByRole` won't find the text content. That was the solution I found working, though perhaps we could access it different way